dev/source/docs/opendroneid.rst

```diff
@@ -0,0 +1,122 @@
+
+Future enhancements are planned.
+
+For an example see the hwdef and param files for an `ODID version of CubeOrange <https://github.com/ArduPilot/ardupilot/tree/master/libraries/AP_HAL_ChibiOS/hwdef/CubeOrange-ODID>`__ with a CAN OpenDroneID module integrated on the carrier board.
+
+.. note:: that the board ID has been changed different from normal firmware, to prevent loading it on a remoteid equipped system by a GCS. And the DID params have been set to read-only to prevent users from changing them and defeating the system.
+
+Building Firmware for Bench Testing and Experimentation
+=======================================================
+
+An experimental firmware version can be built for any autopilot by:
+
+- Adding the line "define AP_OPENDRONEID_ENABLED 1" to its hwdef.dat file, or simply building with the  waf build option, ``--enable-opendroneid``
+- Adding the waf build option, ``--enable-firmware-checking``, will add another tamper proofing check. If enabled, a bootloader will not run firmware with a different board ID. This extends the protection already afforded by using a unique
+
+
+Testing
+=======
+
+- Update MAVProxy/pymavlink to at least 1.8.54 (https://ardupilot.org/mavproxy/docs/getting_started/download_and_installation.html#updating)
+- In the directory you will be running MAVProxy from, create a remoteid.scr file with this content:
+
+.. code-block::
+
+    module load OpenDroneID
+
+    opendroneid set UAS_ID_type 1
+    opendroneid set UA_type 1
+    opendroneid set area_ceiling 700
+    opendroneid set area_count 1
+    opendroneid set area_floor -200
+    opendroneid set area_radius 1000
+    opendroneid set category_eu 1
+    opendroneid set class_eu 1
+    opendroneid set classification_type 1
+    opendroneid set description "TestDrone1"
+    opendroneid set description_type 1
+    opendroneid set operator_id "TestPilot1"
+    opendroneid set operator_id_type 1
+    opendroneid set operator_location_type 0      #note that a value of 1 is usually required in actual operation. This requires using a GCS with its own GPS for testing.
+    opendroneid set rate_hz 1
+
+This sets up the required parameters a user would normally have to enter into the RemoteID system from a GCS.
+
+- Attach autopilot with an OpenDroneID compatible Remote ID module , `like this <https://github.com/ArduPilot/ArduRemoteID>`__ , attached to one of its serial ports to PC and start MAVProxy with the console and map with the option to set the operator location to the vehicle arming location:
+
+.. code::
+
+  MAVProxy.py --console --map
+
+.. note:: this may not be legal in some jurisdictions in actual use, but the OpenDroneID module firmware will reject this assignment with a suitable build option set in its firmware for those jurisdictions. This allows testing on PCs without a GPS attached to determine operator location.
+
+- Make sure the autopilot serial port that the module is attached is setup for 57.6K and MAVLink2 
```

Comment:
only if using a UART right? Would not apply if using DroneCAN?

Reply:
right...adding info on that on that line...thanks